cmd/manager/main.go

```diff
@@ -144,85 +155,143 @@ func newRootCommand() *cobra.Command {
 			hiveNSName := utils.GetHiveNamespace()
 			log.Infof("hive namespace: %s", hiveNSName)
 
-			// Allow an env var to disable leader election, useful when testing from source with make run to eliminate the wait time.
-			var disableLeaderElection bool
-			val := os.Getenv("DISABLE_LEADER_ELECTION")
-			if val != "" {
-				disableLeaderElection, err = strconv.ParseBool(val)
-				if err != nil {
-					log.WithField("value", val).Fatal("Error parsing DISABLE_LEADER_ELECTION env var")
+			// Create and start liveness and readyness probe endpoints
+			http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusOK)
+			})
+			http.HandleFunc("/readyz", func(w http.ResponseWriter, r *http.Request) {
+				if isLeader {
```

Comment:
Does `http.HandleFunc` provide a happens-before edge that will guarantee that changes to this field made in other goroutines will be visible in this goroutine?

Reply:
I think yes. Otherwise readiness probe would have continuously failed during my testing even after the pod gets elected as a leader.

Reply:
I think no. Just because it happens almost all of the time, does not make it a guarantee. You cannot write a pure int in one goroutine and the value to be seen in another goroutine. You need some mechanism to create a happens-before edge to guarantee visibility.